tests/compile-fail/cast_fn_ptr_unsafe.rs

```diff
@@ -0,0 +1,10 @@
+// just making sure that fn -> unsafe fn casts are handled by rustc so miri doesn't have to
```

Comment:
I'm confused by these two tests. Whether `fn`s can be casted to `unsafe fn` is to some extent irrelevant to Miri - they can still always be transmuted. Miri has no safe/unsafe distinctions.